realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1415,8 +1421,9 @@ private void checkMaxDepth(int maxDepth) {
      * version, nothing will happen.
      *
      * @param configuration {@link RealmConfiguration}
+     * @throws FileNotFoundException if the Realm file doesn't exist.
      */
-    public static void migrateRealm(RealmConfiguration configuration) {
+    public static void migrateRealm(RealmConfiguration configuration) throws FileNotFoundException {
```

Comment:
Thinking about this a bit more. FileNotFoundException is a lot better than `IllegalArgument/IllegalState`, and perhaps it is good to force people to consider that the file they try to migrate doesn't exist. I think I could be convinced this is OK :+1: